src/compiler/transformers/map-imports-to-path-aliases.ts

```diff
@@ -0,0 +1,60 @@
+import { normalize, relative } from 'path';
+import ts from 'typescript';
+import type * as d from '../../declarations';
+
+/**
+ * This method is responsible for replacing user-defined import path aliases ({@link https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping})
+ * with generated relative import paths during the transformation step of the TS compilation process.
+ * This action is taken to prevent issues with import paths not being transpiled at build time resulting in
+ * unknown imports in output code for some output targets (`dist-collection` for instance). Output targets that do not run through a bundler
+ * are unable to resolve imports using the aliased path names and TS intentionally does not replace resolved paths as a part of
+ * their compiler ({@link https://github.com/microsoft/TypeScript/issues/10866})
+ *
+ * @param config The Stencil configuration object.
+ * @returns A factory for creating a {@link ts.Transformer}.
+ */
+export const mapImportsToPathAliases = (config: d.Config): ts.TransformerFactory<ts.SourceFile> => {
+  return (transformCtx) => {
+    let sourceFile: string;
```

Comment:
Yes this is needed. Using the approach you suggested errors out. `node.getSourceFile()` returns `undefined`

Reply:
it looks to me like you have it up here because you can't get the right value until the function returned here is called:

https://github.com/ionic-team/stencil/blob/88a6974882c23e97fbd257e81cd8b1349235345e/src/compiler/transformers/map-imports-to-path-aliases.ts#L83-L79

and then you need to use it within the `visit` body. I think doing `let foo: string;` does always create a `strictNullChecks` error though because since it's not initialized with a string the variable is `undefined` (and therefore not a string).

One alternative approach could be to curry the `sourceFile` argument like this:

```ts
const visit = (sourceFile: string) => (node: ts.Node): ts.VisitResult<ts.Node> => {
```

then below you could do something like

```ts
    return (tsSourceFile) => {
      const sourceFile = tsSourceFile.fileName;

      return ts.visitEachChild(tsSourceFile, visit(sourceFile), transformCtx);
    };
```

which would I think fix the `strictNullChecks` error and is arguably "more correct" code since we don't declare a variable without initializing it and then access it later w/ the implicit assumption it's been assigned in the intervening time

Reply:
This worked! Thanks for the suggestion!